components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaService.java

```diff
@@ -129,6 +136,76 @@ public Info getInfo() {
         return tags;
     }
 
+    public ArrayList<String> getImportModules() {
+        return importModules;
+    }
+
+    /**
+     * To set the names of all the modules to be imported from the Ballerina Central
+     *
+     * @param moduleName     The name of the module which is stored in Ballerina Central
+     */
+    public void setImportModules(String moduleName) {
+        while (!this.importModules.contains(moduleName)) {
+            this.importModules.add(moduleName);
+        }
+        importModules.removeAll(Collections.singletonList(null));
+    }
+
+    /**
+     * To extract the ballerina module name from the openAPI definitions, if the api level interceptors are to be
+     * accessed from the Ballerina Central
+     *
+     * @param openAPI       {@link OpenAPI} object
+     */
+    public void extractImportModules (OpenAPI openAPI) {
+        // Regular Expression which indicates the Ballerina Module
+        String moduleRegEx = "\\w*" + "/" + "\\w*";
+
+        ArrayList<String> requestInterceptorStatement = new ArrayList<>();
+        ArrayList<String> requestInterceptorModule = new ArrayList<>();
+
+        Optional<Object> apiRequestInterceptor = Optional.ofNullable(openAPI.getExtensions()
+                .get(OpenAPIConstants.REQUEST_INTERCEPTOR));
+
+        if (apiRequestInterceptor.toString().contains(OpenAPIConstants.BALLERINA_CENTRAL_KEYWORD)) {
+            requestInterceptorStatement.add(apiRequestInterceptor.toString());
+            requestInterceptorStatement.forEach(statement -> {
```

Comment:
`requestInterceptorStatement` always going to contain only one item. So do we need `forEach`. Or do we need `requestInterceptorStatement` and `requestInterceptorModule` to be a list in the first place?